data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/GreaterThanOperator.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class GreaterThanOperator implements Operator<Boolean> {
+    @Override
+    public String getSymbol() {
+        return ">";
+    }
+
+    @Override
+    public Boolean eval(Object... args) {
+        checkArgument(args.length == 2, "Operands length needs to be 2.");
+        if ((args[0] instanceof Integer) && (args[1] instanceof Float)) {
+            return ((Integer) args[0]) > ((Float) args[1]);
+        } else if ((args[0] instanceof Float) && (args[1] instanceof Integer)) {
+            return ((Float) args[0]) > ((Integer) args[1]);
+        } else if ((args[0] instanceof Integer) && (args[1] instanceof Integer)) {
+            return ((Integer) args[0]) > ((Integer) args[1]);
+        } else if ((args[0] instanceof Float) && (args[1] instanceof Float)) {
+            return ((Float) args[0]) > ((Float) args[1]);
+        }
+        throw new IllegalArgumentException(args[0] + " or " + args[1] + " should be either Float or Integer");
```

Comment:
This pattern is repeated a lot throughout the operators. Is there a way to break this out into a method with something like the following declaration so it can be reused for all of them? It may end up being ugly anyways given that you can't inject the operator super easily.

```
private evalNumericalOperator(Object... args, String Operator)
```

Reply:
I have also thought upon refactoring it. But it seems to me repeating this snippet of very similar logic cannot be avoided unless operator can be injected to indicate explicit method call that takes in args of generic Number & Comparable type. e.g. Integer.compareTo(Integer b) will not work since it only compares Integer to Integer